src/symbolize.cc

```diff
@@ -590,6 +567,48 @@ OpenObjectFileContainingPcAndGetStartAddress(uint64_t pc,
       return -1;  // Malformed line.
     }
 
+    // Determine the base address by reading ELF headers in process memory.
+    ElfW(Ehdr) ehdr;
+    if (flags_start[0] == 'r' &&
+        ReadFromOffsetExact(mem_fd, &ehdr, sizeof(ElfW(Ehdr)), start_address) &&
+        memcmp(ehdr.e_ident, ELFMAG, SELFMAG) == 0) {
```

Comment:
Could you add a comment for this? I think this is skipping non-text maps for usual cases.

Reply:
It is in fact skipping non-readable maps. I've added a comment.